src/app/code/community/Shopgate/Cloudapi/Model/OAuth2/Db/Pdo.php

```diff
@@ -222,6 +222,29 @@ public function setAuthorizationCode(
         return $stmt->execute(compact('code', 'clientId', 'userId', 'resourceType', 'expires', 'scope'));
     }
 
+    /**
+     * @param string $oldEmail
+     * @param string $newEmail
+     */
+    public function updateEmailTokens($oldEmail, $newEmail)
+    {
+        $stmt = $this->db->prepare(
+            $sql = sprintf(
+                'UPDATE %s SET user_id=:newEmail WHERE user_id=:oldEmail',
+                $this->config['refresh_token_table']
+            )
+        );
+        $stmt->execute(compact('newEmail', 'oldEmail'));
+
+        $stmt = $this->db->prepare(
+            $sql = sprintf(
+                'UPDATE %s SET user_id=:newEmail WHERE user_id=:oldEmail',
+                $this->config['access_token_table']
+            )
+        );
+        $stmt->execute(compact('newEmail', 'oldEmail'));
```

Comment:
I reviewed it a bit further, this is a bit dangerous. You'll need to figure out the user's current token hash and use that as a WHERE clause. This sql modifies all entries, which is undesired as a user may have multiple keys on different shops/websites.